{"name": "test_login[2072583546-np123456-True]", "status": "failed", "statusDetails": {"message": "Failed: 登录成功断言失败，未找到主页元素: Message: An element could not be located on the page using the given search parameters.; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nNoSuchElementError: An element could not be located on the page using the given search parameters.\n    at AndroidUiautomator2Driver.findElOrEls (C:\\Users\\lv\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\appium-android-driver\\lib\\commands\\find.ts:86:11)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at AndroidUiautomator2Driver.findElOrElsWithProcessing (C:\\Users\\lv\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\basedriver\\commands\\find.ts:60:12)\n    at AndroidUiautomator2Driver.findElement (C:\\Users\\lv\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\basedriver\\commands\\find.ts:75:12)", "trace": "self = <flutter_app_test.testcases.test_login.TestLogin object at 0x000001B62B15ED50>, username = '2072583546', password = 'np123456'\nexpected = True\n\n    @pytest.mark.parametrize(\"username,password,expected\", [\n        (\"2072583546\", \"np123456\", True),\n        # (\"invalid_user\", \"invalid_pass\", False),\n    ])\n    def test_login(self, username, password, expected):\n        login_page = LoginPage(self.driver)\n        login_page.login(username, password)\n        if expected:\n            # 登录成功断言\n            try:\n>               assert self.driver.find_element(AppiumBy.ID, \"com.newpay.merchant:id/home\").is_displayed()\n\ntestcases\\test_login.py:50: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n..\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:898: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n..\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:429: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <appium.webdriver.errorhandler.MobileErrorHandler object at 0x000001B62B298980>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"An element could not be located on the page u...ode_modules\\\\\\\\appium\\\\\\\\node_modules\\\\\\\\@appium\\\\\\\\base-driver\\\\\\\\lib\\\\\\\\basedriver\\\\\\\\commands\\\\\\\\find.ts:75:12)\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"\n        https://www.w3.org/TR/webdriver/#errors\n        \"\"\"\n        payload = response.get('value', '')\n        if isinstance(payload, dict):\n            payload_dict = payload\n        else:\n            try:\n                payload_dict = json.loads(payload)\n            except (json.JSONDecodeError, TypeError):\n                return\n            if not isinstance(payload_dict, dict):\n                return\n        value = payload_dict.get('value')\n        if not isinstance(value, dict):\n            return\n        error = value.get('error')\n        if not error:\n            return\n    \n        message = value.get('message', error)\n        stacktrace = value.get('stacktrace', '')\n        # In theory, we should also be checking HTTP status codes.\n        # Java client, for example, prints a warning if the actual `error`\n        # value does not match to the response's HTTP status code.\n        exception_class: Type[sel_exceptions.WebDriverException] = ERROR_TO_EXC_MAPPING.get(\n            error, sel_exceptions.WebDriverException\n        )\n        if exception_class is sel_exceptions.WebDriverException and message:\n            if message == 'No such context found.':\n                exception_class = appium_exceptions.NoSuchContextException\n            elif message == 'That command could not be executed in the current context.':\n                exception_class = appium_exceptions.InvalidSwitchToTargetException\n    \n        if exception_class is sel_exceptions.UnexpectedAlertPresentException:\n            raise sel_exceptions.UnexpectedAlertPresentException(\n                msg=message,\n                stacktrace=format_stacktrace(stacktrace),\n                alert_text=value.get('data'),\n            )\n>       raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))\nE       selenium.common.exceptions.NoSuchElementException: Message: An element could not be located on the page using the given search parameters.; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       NoSuchElementError: An element could not be located on the page using the given search parameters.\nE           at AndroidUiautomator2Driver.findElOrEls (C:\\Users\\lv\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\appium-android-driver\\lib\\commands\\find.ts:86:11)\nE           at processTicksAndRejections (node:internal/process/task_queues:105:5)\nE           at AndroidUiautomator2Driver.findElOrElsWithProcessing (C:\\Users\\lv\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\basedriver\\commands\\find.ts:60:12)\nE           at AndroidUiautomator2Driver.findElement (C:\\Users\\lv\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\basedriver\\commands\\find.ts:75:12)\n\n..\\.venv\\Lib\\site-packages\\appium\\webdriver\\errorhandler.py:125: NoSuchElementException\n\nDuring handling of the above exception, another exception occurred:\n\nself = <flutter_app_test.testcases.test_login.TestLogin object at 0x000001B62B15ED50>, username = '2072583546', password = 'np123456'\nexpected = True\n\n    @pytest.mark.parametrize(\"username,password,expected\", [\n        (\"2072583546\", \"np123456\", True),\n        # (\"invalid_user\", \"invalid_pass\", False),\n    ])\n    def test_login(self, username, password, expected):\n        login_page = LoginPage(self.driver)\n        login_page.login(username, password)\n        if expected:\n            # 登录成功断言\n            try:\n                assert self.driver.find_element(AppiumBy.ID, \"com.newpay.merchant:id/home\").is_displayed()\n            except Exception as e:\n>               pytest.fail(f\"登录成功断言失败，未找到主页元素: {e}\")\nE               Failed: 登录成功断言失败，未找到主页元素: Message: An element could not be located on the page using the given search parameters.; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE               Stacktrace:\nE               NoSuchElementError: An element could not be located on the page using the given search parameters.\nE                   at AndroidUiautomator2Driver.findElOrEls (C:\\Users\\lv\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\appium-android-driver\\lib\\commands\\find.ts:86:11)\nE                   at processTicksAndRejections (node:internal/process/task_queues:105:5)\nE                   at AndroidUiautomator2Driver.findElOrElsWithProcessing (C:\\Users\\lv\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\basedriver\\commands\\find.ts:60:12)\nE                   at AndroidUiautomator2Driver.findElement (C:\\Users\\lv\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\basedriver\\commands\\find.ts:75:12)\n\ntestcases\\test_login.py:52: Failed"}, "parameters": [{"name": "username", "value": "'2072583546'"}, {"name": "password", "value": "'np123456'"}, {"name": "expected", "value": "True"}], "start": 1748513014989, "stop": 1748513031938, "uuid": "b9e6b0ef-60ba-461b-aa42-87f8651af97f", "historyId": "cb29e78f39d4e5d3782070b664c1f744", "testCaseId": "93a39feb8eb701231e6c069b76eeac98", "fullName": "testcases.test_login.TestLogin#test_login", "labels": [{"name": "feature", "value": "登录功能"}, {"name": "parentSuite", "value": "testcases"}, {"name": "suite", "value": "test_login"}, {"name": "subSuite", "value": "TestLogin"}, {"name": "host", "value": "DESKTOP-QVK4EQ1"}, {"name": "thread", "value": "31356-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "testcases.test_login"}]}